lightning/src/ln/onchaintx.rs

```diff
@@ -939,23 +925,21 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 	#[cfg(any(test, feature="unsafe_revoked_tx_signing"))]
 	pub(crate) fn get_fully_signed_copy_holder_tx(&mut self, funding_redeemscript: &Script) -> Option<Transaction> {
 		if let Some(ref mut holder_commitment) = self.holder_commitment {
-			match self.key_storage.sign_holder_commitment(holder_commitment, &self.secp_ctx) {
-				Ok(sig) => {
-					Some(holder_commitment.add_holder_sig(funding_redeemscript, sig))
-				},
-				Err(_) => return None,
-			}
+			let (sig, htlc_sigs) = self.key_storage.sign_holder_commitment(holder_commitment, &self.secp_ctx).expect("sign holder commitment");
+			self.holder_htlc_sigs = Some(Self::extract_holder_sigs(holder_commitment, htlc_sigs));
+			Some(holder_commitment.add_holder_sig(funding_redeemscript, sig))
 		} else {
 			None
 		}
 	}
 
+	// The current holder commitment tx MUST have been signed before calling this with one of its outpoints.
```

Comment:
I don't see how we can guarantee this without changes in ChannelMonitor - you can have multiple ChannelMonitors, one which signed and broadcasted a commitment tx, and another which is now trying to broadcast HTLC transactions without ever having signed the commitment transaction(s).

Reply:
I reworked this to remove this assumption, and also added relevant docs to explain why we might still need to sign the HTLCs separately in some cases.